### PR TITLE
build(ci): add snippets workflow

### PR DESCRIPTION
**Pull request by ryannikolaidis**
We want to trigger the make_mdx script from CI in order to hydrate the documentation snippets from example code in the unstructured library, that way we can keep it up to date.

## Changes

- Add GH Actions workflow that calls the make_mdx script and creates a pull request with the changes. This workflow is triggered both by workflow dispatch and a daily running cron
- Updates the make_mdx script for readability as well as Linux compatibility so that it will execute on the runner.

## Testing

Deleted two snippet files from this branch and then ran the workflow against this branch. Here is that run: https://github.com/Unstructured-IO/docs/pull/10

### CODE REVIEW

File: scripts/make_mdx.sh [hunk 1]
```diff
@@ -1,59 +1,75 @@
...
-}
+if [ $# -ne 2 ]; then
+    echo "Error: Two paths (destination and source) are required."
+    echo "Usage: $0 <destination_directory> <source_directory>"
+    exit 1
+fi
+
+# Convert relative paths to absolute paths and ensure they are valid
+DEST_TARGET_DIR=$(realpath --relative-to=. "$1" 2>/dev/null || echo "$1")
+SRC_TARGET_DIR=$(realpath --relative-to=. "$2" 2>/dev/null || echo "$2")
```
Comment (ryannikolaidis) [inline scripts/make_mdx.sh:14]: no longer need to worry about passing in abs path

File: scripts/make_mdx.sh [hunk 2]
```diff
@@ -1,59 +1,75 @@
...
+
+# Create the directories if they do not exist
+mkdir -p "${DEST_TARGET_DIR}"
+mkdir -p "${SRC_TARGET_DIR}"
+
+# Normalize the paths to absolute paths again after creation
+DEST_TARGET_DIR=$(realpath "$DEST_TARGET_DIR")
+SRC_TARGET_DIR=$(realpath "$SRC_TARGET_DIR")
```
Comment (ryannikolaidis) [inline scripts/make_mdx.sh:22]: no longer need to worry about passing in abs path

File: scripts/make_mdx.sh [hunk 3]
```diff
@@ -1,59 +1,75 @@
...
+REPO_URL="https://github.com/Unstructured-IO/unstructured"
+DEST_PYTHON_PATH="docs/source/ingest/destination_connectors/code/python/"
+DEST_SHELL_PATH="docs/source/ingest/destination_connectors/code/bash/"
+SRC_PYTHON_PATH="docs/source/ingest/source_connectors/code/python/"
+SRC_SHELL_PATH="docs/source/ingest/source_connectors/code/bash/"
+
+# Clone only the necessary directories using sparse-checkout
+git clone --depth 1 --filter=blob:none --sparse "${REPO_URL}" "${WORK_DIR}"
+cd "${WORK_DIR}" || exit
+git sparse-checkout set "${DEST_PYTHON_PATH}" "${DEST_SHELL_PATH}" "${SRC_PYTHON_PATH}" "${SRC_SHELL_PATH}"
```
Comment (ryannikolaidis) [inline scripts/make_mdx.sh:40]: just a little easier to grok and doesn't involve initing an empty space first

**Review by potter-potter — APPROVED** [review]
very nice

File: scripts/make_mdx.sh [hunk 4]
```diff
@@ -1,59 +1,75 @@
...
+    cd "${CODE_DIR}" || exit
+    echo "Converting files in ${CODE_DIR}"
+    for f in *.py *.sh; do
+        if [ -f "$f" ]; then
+            local extension="${f##*.}"
+            local lang="${extension:0:2}"  # Simplified logic for language detection
+            lang=${lang/py/python}        # If Python, set to 'python'
+            lang=${lang/sh/bash}          # If Shell, set to 'bash'
+            # Create .mdx file with correct code fences
+            awk -v lang="$lang" 'BEGIN {print "```" lang}
```
[the remark this answers is not in the document]
Comment (ryannikolaidis) [reply scripts/make_mdx.sh:60]: yep! here's my proposal though: I'm going to update and validate locally, but let's not run the change in this PR. let's merge it with the change and then trigger on main to actually validate the end to end automation in CI

[the remark this answers is not in the document]
Comment (MKhalusova) [reply scripts/make_mdx.sh:60]: Sure. Feel free to validate first

**Pull request merged**